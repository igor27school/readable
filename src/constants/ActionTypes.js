export const CHANGE_SORT_ORDER = "CHANGE_SORT_ORDER"
export const RECEIVE_CATEGORIES = "RECEIVE_CATEGORIES"
export const RECEIVE_ALL_POSTS = "RECEIVE_ALL_POSTS"
export const RECEIVE_CATEGORY_POSTS = "RECEIVE_CATEGORY_POSTS"
export const RECEIVE_POST = "RECEIVE_POST"
export const RECEIVE_POST_COMMENTS = "RECEIVE_POST_COMMENTS"
export const RECEIVE_VOTE = "RECEIVE_VOTE"
export const ADD_POST = "ADD_POST"
export const ADD_COMMENT = "ADD_COMMENT"
export const REMOVE_OBJECT = "REMOVE_OBJECT"
export const RECEIVE_COMMENT = "RECEIVE_COMMENT"
export const MODIFY_COMMENT = "MODIFY_COMMENT"
export const MODIFY_POST = "MODIFY_POST"
