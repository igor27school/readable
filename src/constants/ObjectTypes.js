// Identifies which object the operation is being performed on.
// Used for adding, deleting, editing, voting.
export const POST_TYPE = 'POST_TYPE'
export const COMMENT_TYPE = 'COMMENT_TYPE'
