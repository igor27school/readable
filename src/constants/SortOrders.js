export const SORT_BY_SCORES = "SORT_BY_SCORES"
export const SORT_BY_TIMESTAMPS = "SORT_BY_TIMESTAMPS"
