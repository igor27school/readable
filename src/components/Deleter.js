import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { COMMENT_TYPE, POST_TYPE } from '../constants/ObjectTypes'
import { deleteObject } from '../actions/ActionCreators'

/**
* @description This component is used in the PostDetails and PostSummary views, as well as Comment component. It enables the user to Delete a given post or comment.
*/
export class Deleter extends Component {
  static propTypes = {
    objectType: PropTypes.oneOf([COMMENT_TYPE, POST_TYPE]).isRequired,
    id: PropTypes.string.isRequired,
    deleteObject: PropTypes.func.isRequired,
    history: PropTypes.object,
  }
  handleClick = () => {
    const {objectType, id, deleteObject, history} = this.props
    deleteObject(objectType, id)
    if (history) {
      history.push('/')
    }
  }
  render() {
    return (
      <button onClick={this.handleClick}>Remove</button>
    )
  }
}

function mapDispatchToProps (dispatch) {
  return {
    deleteObject: (objectType, id) => dispatch(deleteObject(objectType, id)),
  }
}

export default connect(null, mapDispatchToProps)(Deleter)
