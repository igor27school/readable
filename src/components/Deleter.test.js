import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import { COMMENT_TYPE } from '../constants/ObjectTypes'
import { Deleter } from './Deleter'

jest.mock('../utils/ServerAPI')

describe('Deleter tests', () => {
  const setup = () => {
    const props = {
      objectType: COMMENT_TYPE,
      id: 'testComment',
      deleteObject: jest.fn(),
    }
    const renderer = new ShallowRenderer()
    return {
      props,
      renderer
    }
  }

  it('Deleter matches the snapshot', () => {
    const { renderer, props } = setup()
    renderer.render(<Deleter {...props}/>)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
