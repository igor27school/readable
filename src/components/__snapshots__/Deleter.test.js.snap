// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deleter tests Deleter matches the snapshot 1`] = `
<button
  onClick={[Function]}
>
  Remove
</button>
`;
